### ISSUE TITLE
Base Attack Speed at Level 1 is not included in Static Data API

### ISSUE THREAD

**Issue by lmcdougald** [RiotGames/developer-relations#53]
With changes to the jungle with Runes Reforged (about 1/3 way through 2018 due to dumb cases like Sejuani taking Fleet Footwork), Riot added "attack speed at level 1" to a number of junglers to help their clear speed. Some include Sejuani, Gragas, Ekko, and more I can't remember right now.

This data is not available through the static API. Please add the necessary field.

**Comment by Linnun** [comment]
Afaik there is no more static data endpoint. Where did you encounter this problem? Could it be ddragon-related?

**Comment by lmcdougald** [comment]
https://developer.riotgames.com/static-data.html

If this is deprecated it hasn't been marked as such.

Yes, it's ddragon-related, by nature of being static API-related.

**Comment by Linnun** [comment]
Please note that the documentation you linked is heavily outdated. There used to be static-data endpoints in the Riot Api, but they have been removed because they were almost identical to data from ddragon, and it was unneccessary work to host it on the Riot Api aswell.

Please also note that the DevRel team does not own ddragon.

**Comment by lmcdougald** [comment]
As best as I understand, developer relations is as close as I'm going to get to a group that can communicate with the team that owns ddragon, so the issue is still applicable to the developer relations group by proxy.

If there is a channel to communicate with the ddragon team, it hasn't been advertised.

**Comment by RiotTuxedo** [comment]
The information in data dragon is sufficient to calculate a champions attack speed at any level. Specifically, *attackspeed* indicates the base attacks per second and *attackspeedperlevel* is the percentage growth statistic that I'll describe in detail below. Let's take Aatrox, as an example:

`"Aatrox": {
  "version": "9.2.1",
  "id": "Aatrox",
  "key": "266",
  ...
  "stats": {
      ...
    "attackspeedperlevel": 2.5,
    "attackspeed": 0.651
  }
}`
http://ddragonexplorer.com/cdn/9.2.1/data/en_US/champion/Aatrox.json

Attack speed per level follows a growth statistic formula such that at level 1 you only get ~70% of the attack speed at lower levels and when you reach higher levels you get 100% of the growth statistic. Here's the formula:

attack speed bonus from levels = growth stat x (level - 1) x (0.7025 + 0.0175 x (level - 1))

for Aatrox level 1 it looks like this:
0.025 x 0 x 0.7025 = 0% bonus attack speed at level 1

for Aatrox level 2 it looks like:
0.025 x (2-1) x (0.7025 + 0.0175 x (2-1)))
0.025 x 1 x (0.7025 + 0.0175)
0.018
1.8% bonus attack speed at level 2

As such, the information necessary to answer any questions related to base attack speed and attack speed per level should be resolved here. Let me know if there are additional questions.

**Comment by meesvandongen** [comment]
> _attackspeed_ indicates the base attacks per second

This is not true, _attackspeed_ indicates the attackspeed at level 1. @lmcdougald is right, there is no way to know the base attack speed.

**Comment by RiotTuxedo** [comment]
Interesting if true. I'll look into it, what's your source for this information?

**Comment by meesvandongen** [comment]
The League Wikia describes it for example in the [Amumu page](https://leagueoflegends.fandom.com/wiki/Amumu). You can look at the champion info box.

**Comment by RiotTuxedo** [comment]
You mentioned the following:

> attackSpeed = (baseAttackspeed * bonusCalculation) + level1Boost

but I've actually observed the following:

> attackSpeed = base attack speed * (bonus calculation + level 1 boost)

Can you confirm? The numbers below indicate the level bonus and attack speed bonuses are additive.

Amumu base attack speed is 0.638
Level 1 Amumu has 0.736 attack speed (+15% attack speed from base)
Level 1 with recurve bow (+25% attack speed) has 0.896 attack speed (+40% attack speed from base)

**Comment by lmcdougald** [comment]
The base attack speed is listed explicitly in patch notes where it was added. Examples include the changes to Gragas, Amumu as mentioned, or even Ekko: 

https://na.leagueoflegends.com/en/news/game-updates/patch/patch-723-notes
https://na.leagueoflegends.com/en/news/game-updates/patch/patch-82-notes
